### PR TITLE
Update konnectivity network proxy server to run as non-root, by defau…

### LINKED ISSUE #101879: konnectivity network proxy tests e2e tests failing
#### Which jobs are failing:

https://k8s-testgrid.appspot.com/sig-api-machinery-network-proxy#ci-kubernetes-e2e-gci-gce-network-proxy-http-connect
https://k8s-testgrid.appspot.com/sig-api-machinery-network-proxy#ci-kubernetes-e2e-gci-gce-network-proxy-grpc

#### Anything else we need to know:

Konnectivity network proxy tests have been failing since 5/7/2021.

Looking at the diff of changes it seems like it's related to running the control plane as non root. (https://github.com/kubernetes/kubernetes/compare/1f3fd1cb8...ca0c04e4d?)

cc @vinayakankugoyal, is there anything that needs to be changed on the konnectivity network proxy to support this? The proxy component communicates with the apiserver with a UDS socket.

This is the error we're seeing in the tests:

```
 error dialing backend: dial unix /etc/srv/kubernetes/konnectivity-server/konnectivity-server.socket: connect: permission denied
```

/sig api-machinery
/sig cloud-provider

### PR DESCRIPTION
**Pull request by vinayakankugoyal**
…lt in kube-up.

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
In PR https://github.com/kubernetes/kubernetes/pull/101164 we updated all core control-plane components to run as non-root in kube-up. We missed updating konnectivity-server which also run as on the control plane and communicates with kube-apiserver over uds.

#### Which issue(s) this PR fixes:

Fixes #101879

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by vinayakankugoyal** [comment]
@Jefftree - the failures seem unrelated to my changes. Could you please confirm? Thanks!

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -1935,6 +1935,29 @@ function prepare-konnectivity-server-manifest {
...
+  if [[ -n "${KONNECTIVITY_SERVER_RUNASUSER:-}" && -n "${KONNECTIVITY_SERVER_RUNASGROUP:-}" && -n "${KONNECTIVITY_SERVER_SOCKET_WRITER_GROUP:-}" ]]; then
+    sed -i -e "s@{{ *run_as_user *}}@runAsUser: ${KONNECTIVITY_SERVER_RUNASUSER}@g" "${temp_file}"
+    sed -i -e "s@{{ *run_as_group *}}@runAsGroup: ${KONNECTIVITY_SERVER_RUNASGROUP}@g" "${temp_file}"
+    sed -i -e "s@{{ *supplemental_groups *}}@supplementalGroups: [${KUBE_PKI_READERS_GROUP}]@g" "${temp_file}"
+    sed -i -e "s@{{ *container_security_context *}}@securityContext:@g" "${temp_file}"
+    sed -i -e "s@{{ *capabilities *}}@capabilities:@g" "${temp_file}"
+    sed -i -e "s@{{ *drop_capabilities *}}@drop: [ ALL ]@g" "${temp_file}"
+    sed -i -e "s@{{ *disallow_privilege_escalation *}}@allowPrivilegeEscalation: false@g" "${temp_file}"
+    mkdir -p /etc/srv/kubernetes/konnectivity-server/
+    chown "${KONNECTIVITY_SERVER_RUNASUSER}":"${KONNECTIVITY_SERVER_RUNASGROUP}" /etc/srv/kubernetes/konnectivity-server
```
Comment (cheftako) [inline cluster/gce/gci/configure-helper.sh:1947]: Directory may already exist and have contents. (kubeconfig and UDS socket). Should be safe (and for the UDS SOCKET necessary) to switch owner. Would suggest chown -R

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -1935,6 +1935,29 @@ function prepare-konnectivity-server-manifest {
...
+    chmod g+w /etc/srv/kubernetes/konnectivity-server
+    if [[ -f /etc/srv/kubernetes/konnectivity-server/konnectivity-server.socket ]]; then
```
Comment (cheftako) [inline cluster/gce/gci/configure-helper.sh:1949]: Should not be necessary to rm if the permissions are right. Proxy-server will attempt to delete this when it starts. Also not a problem if you delete it, as long as the Proxy-server has not started yet. I still think its safer to leave the sockets lifecycle in one place.

File: cluster/gce/util.sh [hunk 3]
```diff
@@ -1367,6 +1367,9 @@ KUBE_PKI_READERS_GROUP: 2005
 ETCD_RUNASUSER: 2006
 ETCD_RUNASGROUP: 2006
 KUBE_POD_LOG_READERS_GROUP: 2007
+KONNECTIVITY_SERVER_RUNASUSER: 2008
+KONNECTIVITY_SERVER_RUNASGROUP: 2008
+KONNECTIVITY_SERVER_SOCKET_WRITER_GROUP: 2008
```
Comment (cheftako) [inline cluster/gce/util.sh:1372]: Do we want to overload 2008? I realize it should just be the KAS and Proxy-Server which have this group but this means we are giving the KAS access to everything secured by the Konnectivity Server group.

Comment (vinayakankugoyal) [reply cluster/gce/util.sh:1372]: Wouldn't konnectivity-server need to support taking a flag to specify which group should have ownership to the socket?
If konnectivity-server runs as lets say user 100 and group 100 and creates the socket if it does not exist then, the permissions of  the socket would be 770 100(user) 100(group). Now konnectivity-server would need to chgrp or similar to change the ownership to another group. I don't think that konnectivity-server supports that today.

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (vinayakankugoyal) [reply cluster/gce/gci/configure-helper.sh]: Done.

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (vinayakankugoyal) [reply cluster/gce/gci/configure-helper.sh]: Done.

File: cluster/gce/util.sh (hunk 3 above)
Comment (cheftako) [reply cluster/gce/util.sh:1372]: Fair.

**Pull request merged**